engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -68,12 +70,27 @@ object PartialWorkflowSources {
       // workflow source
       val wdlSource = getStringValue(WdlSourceKey)
       val workflowSource = getStringValue(WorkflowSourceKey)
-      val wdlSourceWarning = wdlSource.map(_ => wdlSourceDeprecationWarning)
+
+      def deprecationWarning(out: String, in: String)(actual: String): String = {
```

Comment:
ToL: It'd be nice if this also included a `(..., expiration: String)` so we can be more exact, eg `"... will be removed from future versions of Cromwell as of July 2018"`

Reply:
If we do that we should also create a ticket to actually remove support for it or it'll very likely be forgotten

Reply:
I'd consider that a PO call since we could be breaking existing installations. Paging @katevoss.

Reply:
Good point, I was thinking we'd at least know "ah yes, now is a fine time to remove it" next time we happened to look at it.

Reply:
Thanks for the page, @mcovarr. If I understand the issue, we are deprecating something and want to add a date of deprecation. After that date, we want to remove it, which would break existing installations. 
As long as we're not breaking anything for 30, could someone make the ticket and we can discuss implications and a plan of action there.

Reply:
Makes me remember that the PEM files are "deprecated" in Cromwell since April... might be time to cut the cord 😄

Reply:
@katevoss see here: https://github.com/broadinstitute/cromwell/issues/2985